src/components/Header.js

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { useHistory, Link } from 'react-router-dom';
+import { AppBar, Toolbar, Typography, Button, Menu, MenuItem } from '@material-ui/core';
+import NestedMenuItem from "material-ui-nested-menu-item";
+import { setLocationToUrl } from '../util';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  menuButton: {
+    marginRight: theme.spacing(2),
+  },
+  title: {
+    flexGrow: 1,
+  },
+}));
+
+export default function Header(props) {
+  const classes = useStyles();
+	const [anchorEl, setAnchorEl] = React.useState(null);
+	const [menuPosition, setMenuPosition] = useState(null);
+	const history = useHistory();
+
+	const handleClick = (event) => {

```

Comment:
Make use of ```useCallback``` hook here to avoid the creation of a new copy of this function every time this component re renders.